### ISSUE TITLE
Change LatoFont as optional external dependency

### ISSUE THREAD

**Issue by mojtabakaviani** [QuestPDF/QuestPDF#913]
The main package of QuestPDF includes the Lato font as an external dependency, which is copied to the output during the build process even if it is not used in the PDF files. It might be more efficient to make this feature optional or to create a separate package that includes the Lato font.

**Comment by MarcinZiabek** [comment]
Thank you for sharing your feedback.

Indeed, it could be beneficial to distribute the font files within a separate nuget. They don't increase the nuget package file significantly, but may slightly increase the application deployment package. 

Given that this is a breaking change, it's crucial that we approach it with careful thought and preparation. I'm eager to hear the perspectives of other library users.

**Comment by fdcastel** [comment]
I too was surprised to find a `LatoFont/` folder in my app output.

There is a way to remove this?

**Comment by fdcastel** [comment]
Workaround: add

```xml
  <!-- Remove LatoFont (from QuestPDF) -->
  <Target Name="RemoveLatoFontAfterPublish" AfterTargets="Publish">
    <RemoveDir Directories="$(PublishDir)\LatoFont" />
  </Target>
```

to the end of your `.csproj`.

You may want to use `$(OutDir)` instead of `$(PublishDir)`.

**Comment by ariacom** [comment]
Hi, same remark for me, I am surprised to see a new LatoFont shortcut folder in all my projects:
![image](https://github.com/QuestPDF/QuestPDF/assets/12451331/607a3a18-6f92-404f-a4b5-c22c42a92333)
I had to understand that it comes from QuestPDF... 
do you have an update to make this transparent ?

**Comment by mojtabakaviani** [comment]
Like other libraries such as SkiaSharp, create an asset package that provides fonts for everyone in need.

**Comment by Adityageek** [comment]
Hi can we keep LatoFont folder inside QuestPDF packages only?

**Comment by MarcinZiabek** [comment]
> Like other libraries such as SkiaSharp, create an asset package that provides fonts for everyone in need.

Considering the history and reasons for including the Lato font in QuestPDF, I believe it is safe to publish it in a separate package. Initially, Lato was meant to be a default font with a limited set of glyphs to avoid artificially enlarging produced PDF files. However, now that QuestPDF supports font subsetting, any desired font can be used without this concern.

The only thing that worries me is that this change is a breaking change. QuestPDF should likely still use Lato as the default font but throw an exception when it is not available. What do you think?


> Hi can we keep LatoFont folder inside QuestPDF packages only?

I will investigate what is possible in this matter. Unfortunately, the build system does not always comply with official .NET documentation, and certain features behave inconsistently across different environments.

I believe that hiding the `LatoFont` folder could be a valid alternative for the majority of users. What do you think?

**Comment by Adityageek** [comment]
The only thing that worries me is that this change is a breaking change. QuestPDF should likely still use Lato as the default font but throw an exception when it is not available. What do you think?  - Yes it is throwing an exception so creates issue for existing user

I believe that hiding the LatoFont folder could be a valid alternative for the majority of users. What do you think? - For the time being this sounds correct option in later stage we can improve this

**Comment by fdcastel** [comment]
> Given that this is a breaking change, it's crucial that we approach it with careful thought and preparation. I'm eager to hear the perspectives of other library users.

I can see two options:

1) Release a new **major** version of `QuestPDF` without LatoFont. And create a new `QuestPDF.LatoFont` with the font assets.

2) Create a new `QuestPDF.Core` -- without LatoFont assets -- and let `QuestPDF` be as it is now. 
    This can be made right now, without a new major version.

From the comments until now, I believe that 1) is the best path forward.

**Comment by ariacom** [comment]
I went back to version 2024.3.1 to avoid the issue, wait for future releases...

**Comment by fdcastel** [comment]
> I went back to version 2024.3.1 to avoid the issue, wait for future releases...

Wait, this is a **very recent** change, then?

In this case, I take back what I said. It should be considered a bug. Not a feature.

Also, I didn't realize the project does not use semver. So the "major version" term is somewhat blurred.

I would say to remove the font assets ASAP.

**Comment by MarcinZiabek** [comment]
> Wait, this is a very recent change, then?

Not quite. The Lato font is part of the package for at least a year. Only the way of how it is included has changed to support older frameworks. For now, I am planning to hide it from the solution/project tree in IDEs. The next step is to find proper strategy to remove it from the package, as it no longer serves its purpose.

> Also, I didn't realize the project does not use semver (using instead a "release date"-based version, which is fine). So the "major version" term is somewhat blurred.

It is a hybrid: `year.month.path`. More or less, it follows a similar convention as JetBrains product versioning, just assuming up to 12 releases within a year. Good or bad, it is like it is, and there is no point in changing it at this moment of library adoption. If I could move back, I would strongly reconsider using semver...

**Comment by MarcinZiabek** [comment]
I am happy to share that starting from the 2024.6.0 version, the LatoFont directory will no longer be visible in the IDE solution structure tree. I am also working on a strategy for how to remove the Lato font completely from the QuestPDF library, as that font no longer serves its primary purpose.

If you want to remove the Lato font completely from the application build artifact (e.g., to minimize deployment package size), please follow these instructions: https://www.questpdf.com/concepts/application-optimization.html

@fdcastel Thank you for providing the workaround.

Thank you for being so helpful. Collaborating with you is a joy 😄

**Comment by Roadrunner67** [comment]
> Indeed, it could be beneficial to distribute the font files within a separate nuget. They don't increase the nuget package file significantly, but may slightly increase the application deployment package.
> 

Slightly -> In my case, my Application package grows from ~15,5M to ~55M. That is a remarkable growth for no added value.
Please consider a solution to this

> f you want to remove the Lato font completely from the application build artifact (e.g., to minimize deployment package size), please follow these instructions: https://www.questpdf.com/concepts/application-optimization.html

This is ineffective when you have     
```
    <PackAsTool>True</PackAsTool>
    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
```
In your .csproj file to generate a packaged tool.

**Comment by znakeeye** [comment]
I tried the [QuestPDF_DoNotIncludeLatoFont_AfterBuild](https://www.questpdf.com/api-reference/text/font-management.html#removing-the-default-lato-font) trick, but the `LatoFont` folder is still copied to the `OutDir` on build. What's the proper way to get rid of it?

**Comment by DawidRobertNowak** [comment]
Hello,

Thank you all for the detailed discussion here.

As of version 2026.9.0, the bundled Lato fonts are distributed as a single compressed archive instead of the `LatoFont` folder: `QuestPDF.Fonts.Lato.br` on .NET 5 and newer, and `QuestPDF.Fonts.Lato.gz` on .NET Standard 2.0 and .NET Framework. This reduces the library's disk footprint by approximately 40%, and the folder no longer appears in the build output or in the IDE solution tree.

Please note that the workarounds posted in this thread, and the ones previously described in our documentation, remove a directory that no longer exists. They now have no effect and can safely be removed from your project files.

Regarding the main request: we have decided not to move the font into a separate package. The bundled fonts now take around 3 MB, and shipping them separately would mean that every user who does not explicitly add the additional package would encounter an exception on the very first document generation. We consider that trade-off unfavourable.

If your deployment package is significantly larger than expected, the cause is most likely the native rendering libraries rather than the font files. During the publish operation, please make sure to target a specific runtime identifier, for example `dotnet publish -r win-x64`. Otherwise the native libraries for all supported platforms are included.

We are therefore closing this issue. Thank you for taking the time to share your use cases — they shaped how the fonts are packaged today.
